# This is free and unencumbered software released into the public domain.

--- auth_credentials.go
+++ auth_credentials.go

@@ -328,9 +328,14 @@
 			t.Actions = append(t.Actions, Action(action))
 		}
 		for _, cred := range ident.Credentials {
+			secretKey, err := os.ReadFile(cred.SecretKey)
+			if err != nil {
+				glog.Warningf("unable to read secret key at %s", cred.SecretKey)
+				return err
+			}
 			t.Credentials = append(t.Credentials, &Credential{
 				AccessKey: cred.AccessKey,
-				SecretKey: cred.SecretKey,
+				SecretKey: string(secretKey),
 			})
 			accessKeyIdent[cred.AccessKey] = t
 		}
